### PR TITLE
Config merge api

### PR DESCRIPTION
**Pull request by Danaorlov1005**

### CODE REVIEW

File: src/app/@ansyn/core/models/config.model.ts [hunk 1]
```diff
@@ -0,0 +1,529 @@
+export interface IConfigModel {
```
Comment (yairtawil) [inline src/app/@ansyn/core/models/config.model.ts]: 1. Put it on ansyn/ansyn
2. Use interfaces for all config's

File: src/app/@ansyn/core/models/fetch-config-providers.ts [hunk 2]
```diff
@@ -1,10 +1,19 @@
...
+import { mergeWith, isArray } from 'lodash';
 
 export const getProviders = (conf): any[] => Object.entries(conf).map(([key, value]): ValueProvider => ({
 	provide: key,
 	useValue: value
 }));
 
-export const fetchConfigProviders = (configPath = 'assets/config/app.config.json') => fetch(configPath)
+export const mergeConfig = (sectionInConfig, wantedMergeChanges) => {
+	if (isArray(sectionInConfig)) {
```
Comment (yairtawil) [inline src/app/@ansyn/core/models/fetch-config-providers.ts]: 1. use Array.isArray( es6 .. )
2. its not array :\

File: src/app/@ansyn/core/models/fetch-config-providers.ts [hunk 3]
```diff
@@ -1,10 +1,19 @@
 import { ValueProvider } from '@angular/core';
+import { mergeWith, isArray } from 'lodash';
 
 export const getProviders = (conf): any[] => Object.entries(conf).map(([key, value]): ValueProvider => ({
 	provide: key,
 	useValue: value
 }));
 
-export const fetchConfigProviders = (configPath = 'assets/config/app.config.json') => fetch(configPath)
+export const mergeConfig = (sectionInConfig, wantedMergeChanges) => {
```
Comment (yairtawil) [inline src/app/@ansyn/core/models/fetch-config-providers.ts]: rename `wantedMergeChanges` to `mergeChanges`

File: src/app/@ansyn/core/models/fetch-config-providers.ts [hunk 4]
```diff
@@ -1,10 +1,19 @@
...
+	if (isArray(sectionInConfig)) {
+		return wantedMergeChanges
+	}
+}
+
+export const fetchConfigProviders = (configPath = 'assets/config/app.config.json', wantedMergeChanges = null) => fetch(configPath)
 	.then(response => response.json())
+	.then(jsonConfig => mergeWith(jsonConfig, wantedMergeChanges, mergeConfig))
```
Comment (yairtawil) [inline src/app/@ansyn/core/models/fetch-config-providers.ts]: mergeConfig function is not necessary, should be removed..

File: src/app/@ansyn/ansyn/config.model.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+import { ILoginConfig } from "../../app/login/models/login.config";
+import { IVisualizersConfig } from "@ansyn/imagery";
+import { IMultipleOverlaysSourceConfig } from "../../../../dist/ansyn/overlays/services/multiple-source-provider";
+import { IIdahoOverlaySourceConfig } from "../../../../dist/ansyn/ansyn/app-providers/overlay-source-providers/idaho-source-provider";
+import { IOpenAerialOverlaySourceConfig } from "./app-providers/overlay-source-providers/open-aerial-source-provider";
+import { IImisightOverlaySourceConfig } from "../../app/imisight/imisight.model";
+import { IPlanetOverlaySourceConfig } from "./app-providers/overlay-source-providers/planet/planet-source-provider";
+import { IAuth0Config } from "../../app/imisight/auth0.service";
+
+export interface IConfigModel {
```
[the remark this answers is not in the document]
Comment (Danaorlov1005) [reply src/app/@ansyn/ansyn/config.model.ts]: ?

File: src/app/@ansyn/ansyn/public_api.ts [hunk 6]
```diff
@@ -9,3 +9,4 @@ export { AnsynComponent } from './ansyn/ansyn.component';
 export { AnsynModule } from './ansyn.module';
 export { IAppState } from './app-effects/app.effects.module';
 export { COMPONENT_MODE } from './app-providers/component-mode';
+export { getProviders } from '../ansyn/fetch-config-providers';
```
Comment (yairtawil) [inline src/app/@ansyn/ansyn/public_api.ts]: ./ instead of ../ansyn

File: src/app/@ansyn/core/public_api.ts [hunk 7]
```diff
@@ -4,7 +4,7 @@ export { AnimatedEllipsisComponent } from './components/animated-ellipsis/animat
 export { forkJoinSafe } from './utils/rxjs/observables/fork-join-safe';
 export { mergeArrays } from './utils/merge-arrays';
 export { selectTime } from './reducers/core.reducer';
-export { fetchConfigProviders } from './models/fetch-config-providers';
+export { fetchConfigProviders } from '../ansyn/fetch-config-providers';
```
Comment (yairtawil) [inline src/app/@ansyn/core/public_api.ts]: remove

**Pull request merged**